aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,486 @@
+
+Navigating child properties
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can also navigate to child properties of your view model. Consider a more complex model class that contains a child ``Address`` property.
+
+.. literalinclude::  forms/sample/final/ViewModels/AddressViewModel.cs
+  :language: c#
+  :lines: 5-16
+  :dedent: 3
+  :emphasize-lines: 1-4,11
+
+In the view, we bind to ``Address.AddressLine1``: 
+
+.. literalinclude::  forms/sample/final/Views/Demo/RegisterAddress.cshtml 
+  :language: HTML
+  :emphasize-lines: 6
+
+The following HTML is generated for `` Address.AddressLine1``:
+
+.. code-block:: HTML
+
+  <input type="text" id="Address_AddressLine1" name="Address.AddressLine1" value="" />
+  
+The Textarea Tag Helper
+-------------------------
+
+The `Textarea Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/TextAreaTagHelper/index.html>`__ tag helper is  similar to the Input Tag Helper.  
+
+- Generates the ``id`` and ``name`` attribute, and the data validation attributes from the model for a `<textarea> <http://www.w3.org/wiki/HTML/Elements/textarea>`__ element. 
+- HTML Helper alternative: ``Html.TextAreaFor``
+
+Consider the following form:
+
+..  literalinclude::  forms/sample/final/Views/Demo/RegisterTextArea.cshtml
+  :language: HTML
+  :emphasize-lines: 4
+
+Using the following model:
+
+.. literalinclude::  forms/sample/final/ViewModels/DescriptionViewModel.cs
+  :language: c#
+  
+The following HTML is generated:
+
+.. code-block:: HTML  
+  :emphasize-lines: 2-8
+  :linenos:
+
+  <form method="post" action="/Demo/RegisterTextArea">
+    <textarea data-val="true" 
+     data-val-maxlength="The field Description must be a string or array type with a maximum length of &#x27;1024&#x27;."
+     data-val-maxlength-max="1024" 
+     data-val-minlength="The field Description must be a string or array type with a minimum length of &#x27;5&#x27;." 
+     data-val-minlength-min="5" 
+     id="Description" name="Description">
+    </textarea>
+    <button type="submit">Test</button>
+    <input name="__RequestVerificationToken" type="hidden" value="<removed for brevity>" />
+  </form>
```

Comment:
@dougbu Either I fixed that and I forgot or I can't reproduce. I'm not seeing any blank lines when I _view _source__ - but I am formatting the output.